addons/account/static/src/components/section_state/section_state_icon.js

```diff
@@ -0,0 +1,39 @@
+import { Component } from '@odoo/owl';
+import { registry } from '@web/core/registry';
+import { standardWidgetProps } from '@web/views/widgets/standard_widget_props';
+
+export const COLLAPSE_ICONS = {
+    collapse_composition: 'fa fa-compress',
+    collapse_prices: 'fa fa-eye-slash',
+}
+
+export class SectionStateIcon extends Component {
+    static template = 'account.SectionStateIcon';
+    static props = {
+        ...standardWidgetProps,
+        iconMapping: Object,
+    };
+
+    get iconClass() {
+        for (const [field, icon] of Object.entries(this.props.iconMapping)) {
+            if (this.props.record.data[field]) {
+                return icon;
+            }
+        }
+
+        return '';
+    }
+}
+
+export const sectionStateIcon = {
+    component: SectionStateIcon,
+    extractProps: ({ options }) => ({
+        iconMapping: {
+            ...COLLAPSE_ICONS,
+            ...options,
```

Comment:
Seems a bit ambiguous to me that the whole field options are considered to be an icon mapping :thinking:
Why don't you provide some hook method and patch the widget to add `is_optional` to the icon mapping ?

You could add a check to see if the field is present before checking its value, and make it clear that some might not be there.